### PR TITLE
[skip-ci][relnotes] Mention behavior change w.r.t. tbb::task_scheduler_init

### PR DESCRIPTION
**Pull request by eguiraud**
Fixes ROOT-11014.

### CODE REVIEW

File: README/ReleaseNotes/v624/index.md [hunk 1]
```diff
@@ -44,6 +44,17 @@ The following people have contributed to this new version:
...
+interfaces in recent TBB versions, as of v6.24 ROOT will not honor a maximum concurrency level set with
+`tbb::task_scheduler_init` but will require instead the usage of `tbb::global_control`:
+
+```cpp
+  //tbb::task_scheduler_init init(2); // does not affect the number of threads ROOT will use anymore
+
+  tbb::global_control c(tbb::global_control::max_allowed_parallelism, 2);
+  ROOT::TThreadExecutor p1;  // will use 2 threads
+  ROOT::TThreadExecutor p2(/*nThreads=*/8); // will still use 2 threads
+```
```
Comment (hageboeck) [inline README/ReleaseNotes/v624/index.md:57]:
```suggestion
`tbb::task_scheduler_init`. Instead, parallelism can be controlled in two ways:
- Set ROOT's parallelism level:
```cpp
  // Method 1:
  ROOT::EnableImplicitMT(2);
  ROOT::TThreadExecutor p1; // ROOT's parallelism level is initialised to 2
  
  // Method 2:
  ROOT::TThreadExecutor p2(8); // ROOT's parallelism level is initialised to 8.
```
The advantage of this method is that other TBB tasks can run concurrently with ROOT's tasks.
- Set TBB's global parallelism level using `tbb::global_control`:
```cpp
  //tbb::task_scheduler_init init(2); // does not affect the number of threads ROOT will use anymore

  tbb::global_control c(tbb::global_control::max_allowed_parallelism, 2);
  ROOT::TThreadExecutor p1;  // will use 2 threads
  ROOT::TThreadExecutor p2(/*nThreads=*/8); // will still use 2 threads
```
```

Comment (hageboeck) [reply README/ReleaseNotes/v624/index.md:57]: Attention:
Github doesn't understand the suggestion, because ``` both delimit code and the suggestion ...

Comment (eguiraud) [reply README/ReleaseNotes/v624/index.md:57]: isn't that more for documentation than for release notes? people look at release notes to see what changed, and the behavior of `ROOT::EnableImplicitMT` didn't change.

I could add a paragraphs in the docs somewhere (where?) about ROOT and parallelism and point to that from the release notes.

Comment (hageboeck) [reply README/ReleaseNotes/v624/index.md:57]: Good idea. Maybe we ask during the team meeting?

File: README/ReleaseNotes/v624/index.md [hunk 2]
```diff
@@ -44,6 +44,17 @@ The following people have contributed to this new version:
...
```
Comment (hageboeck) [inline README/ReleaseNotes/v624/index.md:58]:
```suggestion
Note that it is generally preferred to control ROOT's parallelism using [ROOT::EnableImplicitMT()](https://root.cern/doc/master/namespaceROOT.html#a06f2b8b216b615e5abbc872c9feff40f).
```

**Review by hageboeck — APPROVED** [review]

**Comment by eguiraud** [comment]
@hageboeck better?

**Pull request merged**